### ISSUE TITLE
Refactor API

### ISSUE THREAD

**Issue by lutzzdias** [CaueMaldonadoLima/Manett#14]
Current implementation has no separation of concerns.

The idea is to refactor it into feature-first layers resulting in a structure similar to:

```
api/
├── entries/
│   ├── [id]/
│   │   └── route.ts
│   ├── user/
│   │   └── [id]/
│   │       └── route.ts
│   ├── route.ts
│   ├── application.ts
│   ├── repository.ts
│   └── schema.ts
└── ...
```

Where:
- `route.ts` files act as the controller
- `application.ts` files act as the logic layer
- `repository.ts` files handle db queries (through drizzle)
- `schema.ts` defines the drizzle database schema

**Comment by lutzzdias** [comment]
In the end, the following structure is to be the pattern followed:
```
api/
├── entries/
│   ├── [id]/
│   │   └── route.ts
│   ├── user/
│   │   └── [id]/
│   │       └── route.ts
│   ├── route.ts
│   ├── controller/
│   │   ├── index.ts
│   │   ├── create.ts
│   │   ├── update.ts
│   │   └── remove.ts
│   ├── useCase/
│   │   ├── index.ts
│   │   ├── create.ts
│   │   ├── update.ts
│   │   └── remove.ts
│   ├── repository/
│   │   ├── index.ts
│   │   ├── create.ts
│   │   ├── update.ts
│   │   └── remove.ts
│   ├── docs.md
│   └── schema.ts
├── utils/
│   ├── index.ts
│   ├── auth.ts
│   ├── validation.ts
│   └── ...
├── errors.ts
└── ...
```

Following javascripts highly functional structure, each layer is defined as functions which are then exported (default) by an `index.ts` file. Additionaly, we have an utils folder to allow for cleaner and more readable controllers/useCases by creating helper functions with describable names. Lastly, in the `errors.ts` file we define custom errors which are subclasses of javascript's default `Error` class.

This structure should be repeated on every module / feature of the "backend" of this application.

**Comment by lutzzdias** [comment]
Closed by #19
